File: frontend/src/tourDestinations.js
// Tour destinations for 780.ir. Mirrors backend/internal/searchurl/
// tourdestinations.go — the frontend accumulates conversation context locally
// and redirects without a round trip, so it needs its own copy.
//
// Keyed by the canonical Persian destination name (slots.destination.name).
// The type is data, not something to infer: 780 decides which bucket a
// destination sits in, and it puts کربلا and نجف under INTERNATIONAL.
//
// ⚠️ Never guess a slug for a destination that isn't listed. An unknown slug
// does not 404 on 780 — the tour page spins on a loading state forever with no
// error — so an unlisted destination must produce the unsupported message
// instead of a link.
export const TOUR_DESTINATIONS = {
  // DOMESTIC
  مشهد: { slug: 'mashhad', type: 'DOMESTIC' },
  سرعین: { slug: 'sareyn', type: 'DOMESTIC' },
  ماسال: { slug: 'masal', type: 'DOMESTIC' },
  کیش: { slug: 'kish', type: 'DOMESTIC' },
  سوادکوه: { slug: 'savadkooh', type: 'DOMESTIC' },
  شیرگاه: { slug: 'shirgah', type: 'DOMESTIC' },
  رشت: { slug: 'rasht', type: 'DOMESTIC' },
  رامسر: { slug: 'ramsar', type: 'DOMESTIC' },
  قزوین: { slug: 'ghazvin', type: 'DOMESTIC' },
  ارومیه: { slug: 'orumie', type: 'DOMESTIC' },

  // INTERNATIONAL
  استانبول: { slug: 'istanbul', type: 'INTERNATIONAL' },
  کربلا: { slug: 'karbala', type: 'INTERNATIONAL' },
  آنتالیا: { slug: 'antalya', type: 'INTERNATIONAL' },
  تفلیس: { slug: 'tbilisi', type: 'INTERNATIONAL' },
  ایروان: { slug: 'yerevan', type: 'INTERNATIONAL' },
  باتومی: { slug: 'batumi', type: 'INTERNATIONAL' },
  نجف: { slug: 'najaf', type: 'INTERNATIONAL' },
  دبی: { slug: 'dubai', type: 'INTERNATIONAL' },
}

export function lookupTourDestination(name) {
  return TOUR_DESTINATIONS[name] ?? null
}

// Tours take neither an origin nor a departure date; the results page filters
// by month, not day. So a destination alone is a complete request.
export function buildTourSearchUrl(slots) {
  const dest = lookupTourDestination(slots.destination?.name)
  if (!dest) return null
  return (
    `https://780.ir/tourism/tour/${dest.slug}` +
    `?type=${dest.type}&destinationName=${encodeURIComponent(slots.destination.name)}`
  )
}

File: backend/internal/searchurl/tourdestinations.go
package searchurl

// Tour destinations for 780.ir, from the platform's own zones endpoints
// (tour/v2/zones/DOMESTIC and .../INTERNATIONAL). The raw payloads are kept at
// docs/tour-zones-*.json: unlike every other vocabulary in this repo they are
// NOT re-derivable from anything public — the zones endpoints require
// authorization and no tour sitemap exists — so the source data is committed
// alongside the table it produced.
//
// ⚠️ Never invent a slug for a destination that isn't in this table. A wrong
// slug does not 404: 780's tour page hangs on an infinite loading spinner with
// no error at all, which is a worse dead end than the empty hotel page. An
// unknown destination must be reported as unsupported instead.
//
// The type is part of the data, not something to infer: it selects DOMESTIC vs
// INTERNATIONAL in the URL, and 780 groups a destination one way or the other
// itself (کربلا and نجف are INTERNATIONAL, as they should be).
//
// These lists are 780's *frequent* destinations, not an exhaustive catalogue —
// the endpoint field is literally "destinationFrequency". So coverage is 18
// destinations, and everything else falls through to the unsupported message.

// TourType is the value of the "type" query parameter.
type TourType string

const (
	TourDomestic      TourType = "DOMESTIC"
	TourInternational TourType = "INTERNATIONAL"
)

// TourDestination is one bookable tour destination.
type TourDestination struct {
	Slug string
	Type TourType
}

// tourDestinations is keyed by the canonical Persian name, matching the other
// per-service tables.
//
// Two entries from the source data are deliberately absent:
//   - "وان" (Van, Turkey) — also the everyday word for a bathtub and the
//     spoken form of "one". It is excluded from the international flight table
//     for exactly this reason; including it here would turn "وان حمام رو پر
//     کن" into a tour search.
//   - "پکن+شانگهای" — a combined two-city package whose name isn't a thing
//     anyone says. Mapping the spoken "پکن" to it would silently sell a
//     Beijing+Shanghai package to someone who asked about Beijing.
var tourDestinations = map[string]TourDestination{
	// DOMESTIC
	"مشهد":    {"mashhad", TourDomestic},
	"سرعین":   {"sareyn", TourDomestic},
	"ماسال":   {"masal", TourDomestic},
	"کیش":     {"kish", TourDomestic},
	"سوادکوه": {"savadkooh", TourDomestic},
	"شیرگاه":  {"shirgah", TourDomestic},
	"رشت":     {"rasht", TourDomestic},
	"رامسر":   {"ramsar", TourDomestic},
	"قزوین":   {"ghazvin", TourDomestic},
	"ارومیه":  {"orumie", TourDomestic},

	// INTERNATIONAL
	"استانبول": {"istanbul", TourInternational},
	"کربلا":    {"karbala", TourInternational},
	"آنتالیا":  {"antalya", TourInternational},
	"تفلیس":    {"tbilisi", TourInternational},
	"ایروان":   {"yerevan", TourInternational},
	"باتومی":   {"batumi", TourInternational},
	"نجف":      {"najaf", TourInternational},
	"دبی":      {"dubai", TourInternational},
}

// LookupTourDestination returns the tour destination for a canonical name.
func LookupTourDestination(name string) (TourDestination, bool) {
	d, ok := tourDestinations[name]
	return d, ok
}
